### PR TITLE
Add AWS subnets to AWSCluster and AWSMachineDeployment CRs

### PR DESCRIPTION
**Pull request by pipo02mix**
Request from the customer

### CODE REVIEW

File: cmd/template/cluster/flag.go [hunk 1]
```diff
@@ -91,6 +94,13 @@ func (f *flag) Validate() error {
 			return microerror.Maskf(invalidFlagError, "--%s must only contain alphanumeric characters, and start with a letter", flagClusterID)
 		}
 
+		if f.ControlPlaneSubnet != "" {
+			matchedSubnet, err := regexp.MatchString("^24|25|26|27$", f.ControlPlaneSubnet)
+			if err == nil && !matchedSubnet {
```
Comment (pipo02mix) [inline cmd/template/cluster/flag.go:99]: Can you please point me to which values would be correct

File: cmd/template/nodepool/provider/aws.go [hunk 2]
```diff
@@ -33,6 +34,10 @@ func WriteAWSTemplate(out io.Writer, config NodePoolCRsConfig) error {
 		return microerror.Mask(err)
 	}
 
+	if config.MachineDeploymentSubnet != "" {
+		crs.AWSMachineDeployment.Annotations[annotation.AWSSubnetSize] = config.MachineDeploymentSubnet
+	}
```
Comment (pipo02mix) [inline cmd/template/nodepool/provider/aws.go:39]: I was not sure where to add this logic but I did not find in apiextensions the way to configure it

File: cmd/template/cluster/provider/aws.go [hunk 3]
```diff
@@ -30,6 +32,10 @@ func WriteAWSTemplate(out io.Writer, config ClusterCRsConfig) error {
 		return microerror.Mask(err)
 	}
 
+	if config.ControlPlaneSubnet != "" {
+		crs.AWSCluster.Annotations[annotation.AWSSubnetSize] = config.ControlPlaneSubnet
+	}
+
```
Comment (pipo02mix) [inline cmd/template/cluster/provider/aws.go:38]: I was not sure where to add this logic but I did not find in apiextensions the way to configure it

File: cmd/template/cluster/flag.go (hunk 1 above)
Comment (paurosello) [reply cmd/template/cluster/flag.go:99]: I would say correct values are /20-/25.

26 and 27 not really usable due to subneting

File: cmd/template/cluster/flag.go [hunk 4]
```diff
@@ -91,6 +94,13 @@ func (f *flag) Validate() error {
 			return microerror.Maskf(invalidFlagError, "--%s must only contain alphanumeric characters, and start with a letter", flagClusterID)
 		}
 
+		if f.ControlPlaneSubnet != "" {
+			matchedSubnet, err := regexp.MatchString("^24|25|26|27$", f.ControlPlaneSubnet)
```
Comment (pipo02mix) [inline cmd/template/cluster/flag.go:98]:
```suggestion
			matchedSubnet, err := regexp.MatchString("^20|21|22|23|24|25$", f.ControlPlaneSubnet)
```

File: cmd/template/nodepool/flag.go [hunk 5]
```diff
@@ -112,6 +116,13 @@ func (f *flag) Validate() error {
 		}
 	}
 
+	if f.MachineDeploymentSubnet != "" {
+		matchedSubnet, err := regexp.MatchString("^24|25|26|27$", f.MachineDeploymentSubnet)
```
Comment (pipo02mix) [inline cmd/template/nodepool/flag.go:120]:
```suggestion
		matchedSubnet, err := regexp.MatchString("^20|21|22|23|24|25$", f.MachineDeploymentSubnet)
```

File: cmd/template/cluster/flag.go [hunk 6]
```diff
@@ -91,6 +94,13 @@ func (f *flag) Validate() error {
 			return microerror.Maskf(invalidFlagError, "--%s must only contain alphanumeric characters, and start with a letter", flagClusterID)
 		}
 
+		if f.ControlPlaneSubnet != "" {
+			matchedSubnet, err := regexp.MatchString("^20|21|22|23|24|25$", f.ControlPlaneSubnet)
+			if err == nil && !matchedSubnet {
+				return microerror.Maskf(invalidFlagError, "--%s must be a valid subnet size (24, 25, 26 or 27)", flagControlPlaneSubnet)
```
Comment (pipo02mix) [inline cmd/template/cluster/flag.go:100]:
```suggestion
				return microerror.Maskf(invalidFlagError, "--%s must be a valid subnet size (20, 21, 22, 23, 24 or 25)", flagControlPlaneSubnet)
```

File: cmd/template/nodepool/flag.go [hunk 7]
```diff
@@ -112,6 +116,13 @@ func (f *flag) Validate() error {
 		}
 	}
 
+	if f.MachineDeploymentSubnet != "" {
+		matchedSubnet, err := regexp.MatchString("^20|21|22|23|24|25$", f.MachineDeploymentSubnet)
+		if err == nil && !matchedSubnet {
+			return microerror.Maskf(invalidFlagError, "--%s must be a valid subnet size (24, 25, 26 or 27)", flagMachineDeploymentSubnet)
```
Comment (pipo02mix) [inline cmd/template/nodepool/flag.go:122]:
```suggestion
			return microerror.Maskf(invalidFlagError, "--%s must be a valid subnet size (20, 21, 22, 23, 24 or 25)", flagMachineDeploymentSubnet)
```

**Review by paurosello — APPROVED** [review]

**Pull request merged**